common/app/model/content/Atoms.scala

```diff
@@ -47,35 +47,47 @@ final case class Quiz(
   revealAtEnd: Boolean
 ) extends Atom
 
-
-
+final case class InteractiveAtom(
+  override val id: String,
+  `type`: String,
+  title: String,
+  css: String,
+  html: String,
+  mainJS: Option[String],
+  docData: Option[String]
+) extends Atom
 
 object Atoms extends common.Logging {
+  def extract[T](atoms: Option[Seq[AtomApiAtom]], extractFn: AtomApiAtom => T): Seq[T] = {
+    try {
+      atoms.getOrElse(Nil).map(extractFn)
+    } catch {
+      case e: Exception =>
+        logException(e)
+        Nil
+    }
+  }
+
   def make(content: contentapi.Content): Option[Atoms] = {
     content.atoms.map { atoms =>
-      val quizzes: Seq[atomapi.quiz.QuizAtom] = try {
-        atoms.quizzes.getOrElse(Nil).map(atom => {
-          atom.data.asInstanceOf[AtomData.Quiz].quiz
-        })
-      } catch {
-        case e: Exception =>
-          logException(e)
-          Nil
-      }
-      val media: Seq[MediaAtom] = try {
-        atoms.media.getOrElse(Nil).map(atom => {
-          val id = atom.id
-          val mediaAtom = atom.data.asInstanceOf[AtomData.Media].media
-          MediaAtom.mediaAtomMake(id, mediaAtom)
-        })
-      } catch {
-        case e: Exception =>
-          logException(e)
-          Nil
-      }
+      val quizzes = extract(atoms.quizzes, atom => {
+        val quizAtom = atom.data.asInstanceOf[AtomData.Quiz].quiz
+        Quiz.make(content.id, quizAtom)
+      })
+
+      val media = extract(atoms.media, atom => {
+        val id = atom.id
+        val mediaAtom = atom.data.asInstanceOf[AtomData.Media].media
+        MediaAtom.mediaAtomMake(id, mediaAtom)
+      })
+
+      val interactives = extract(atoms.interactives, atom => {
+        val interactiveAtom = atom.data.asInstanceOf[AtomData.Interactive].interactive
+        InteractiveAtom.make(atom.id, interactiveAtom)
+      })
```

Comment:
Would it make sense to use `partialFunction` here to avoid the `asInstanceOf` and `try/catch` dance here? Something like that:
```
    def make(content: contentapi.Content): Option[Atoms] = {

    def extractAtoms[T](atoms: Option[Seq[AtomApiAtom]], dataToAtom: PartialFunction[AtomData, T]): Seq[T] = atoms.fold(Seq.empty[T])(_.map { atom => dataToAtom(atom.data)})

    val atomDataToQuiz: PartialFunction[AtomData, Quiz] = { case data: AtomData.Quiz => Quiz.make(content.id, data.quiz) }
    val atomDataToMedia: PartialFunction[AtomData, MediaAtom] = { case data: AtomData.Media => MediaAtom.mediaAtomMake(content.id, data.media) }
    val atomDataToInteractive: PartialFunction[AtomData, InteractiveAtom] = { case data: AtomData.Interactive => InteractiveAtom.make(content.id, data.interactive) }

    content.atoms.map { atoms =>
      Atoms(
        quizzes = extractAtoms(atoms.quizzes, atomDataToQuiz),
        media = extractAtoms(atoms.media, atomDataToMedia),
        interactives = extractAtoms(atoms.quizzes, atomDataToInteractive)
      )
    }
  }
```